src/node_crypto.cc

```diff
@@ -84,6 +88,33 @@ using v8::V8;
 using v8::Value;
 
 
+// Subject DER of CNNIC ROOT CA and CNNIC EV ROOT CA are taken from
+// https://hg.mozilla.org/mozilla-central/file/98820360ab66/security/
+// certverifier/NSSCertDBTrustDomain.cpp#l672
+// C = CN, O = CNNIC, CN = CNNIC ROOT
+static const uint8_t CNNIC_ROOT_CA_SUBJECT_DATA[] =
+    "\x30\x32\x31\x0B\x30\x09\x06\x03\x55\x04\x06\x13\x02\x43\x4E\x31\x0E\x30"
+    "\x0C\x06\x03\x55\x04\x0A\x13\x05\x43\x4E\x4E\x49\x43\x31\x13\x30\x11\x06"
+    "\x03\x55\x04\x03\x13\x0A\x43\x4E\x4E\x49\x43\x20\x52\x4F\x4F\x54";
+const uint8_t* cnnic_p = CNNIC_ROOT_CA_SUBJECT_DATA;
+static X509_NAME* cnnic_name =
+    d2i_X509_NAME(nullptr, &cnnic_p, sizeof(CNNIC_ROOT_CA_SUBJECT_DATA));
```

Comment:
Thanks, I missed it. It should be `sizeof(CNNIC_ROOT_CA_SUBJECT_DATA)-1`. Fortunately d2i API handles ASN.1 length properly so that the rest of unnecessary data are ignored. I will fix it.